### ISSUE TITLE
Add 100 credit points everyday somebody uses the system

### ISSUE THREAD

**Issue by wintermeyer** [wintermeyer/animina#250]
The user can earn points by using the system. So the first time he/she uses the system on a given day he/she receives 100 points as a thank you from the system. But only once! Give me one ping Vasily! 😉

https://youtu.be/jr0JaXfKj68?si=cqoIeFkr5RCUVVn8&t=4

PS: If a user uses the system 10 days in a row he/she gets an additional bonus of 250 points.

**Comment by MICHAELMUNAVU83** [comment]
### QUESTION

@wintermeyer  , should we add the 250 bonus points every 10 days or just the first 10 days.

### SUGGESTED IMPLIMENTATION

- I suggest we use the streaks implimentation just like how Snapchat do  , on snapchat , every day we send photos to each other , that counts as a streak and people take pride in having many streaks , A Snapchat Streak (Snapstreak) means you have exchanged (both sent and received) Snaps with a person on your Friends list for at least 3 consecutive “days” (24-hour periods). As long as you keep exchanging at least one set of Snaps each “day,” your streak will keep growing!


- Our implimentation would be similar , we add a `streak` column to the users table that defaults to 0 , each day the user logs in , we add the streak by 1 , if they miss a day , we reset it to 0.


### MARKETING BENEFITS OF THIS

- We can have popups  telling users , 4 more days of using the system , you get 250 points or have ways such that , the higher your streak count , the higher up in the recommendation page you appear , making guys use the system more .
Thoughts?

**Comment by wintermeyer** [comment]
> * I suggest we use the streaks implimentation

Good idea.

> * Our implimentation would be similar , we add a `streak` column to the users table that defaults to 0 , each day the user logs in , we add the streak by 1 , if they miss a day , we reset it to 0.

I don't understand how one column in the users table can help us solving this problem. Please elaborate.

> * We can have popups  telling users , 4 more days of using the system , you get 250 points or have ways such that , the higher your streak count , the higher up in the recommendation page you appear , making guys use the system more .
>   Thoughts?

That was exactly the idea of this concept. A bit more subtile (not in your face) but that is the idea.

*One problem:* I don't expect people to login every day. Or let me rephrase that: They don't logout every day. So they will open their browser and still be logged in. We have to catch that and credit points too in that case.

**Comment by MICHAELMUNAVU83** [comment]
-The column would hold an integer that would increment or reset every day someone uses the system.

**Comment by wintermeyer** [comment]
> The column would hold an integer that would increment or reset every day someone uses the system.

But that is not enough to realise this feature. We need some sort of log or history. Otherwise we don't know if the user was online yesterday or the day before.

Idea: We could misuse `Credit.donor` for that. It has a timestamp. With that we wouldn't even need the extra column in the users table.

**Comment by MICHAELMUNAVU83** [comment]
@wintermeyer  , with your implimentation how would we know if a user has  used the system for 10 days straight?

**Comment by wintermeyer** [comment]
> with your implimentation how would we know if a user has used the system for 10 days straight

Either with a bit of SQL magic or with Elixir. I'd fetch all the Credits which were created by current_user = donor of the last 10 days. If for every day there is one entry I know that the user was active for 10 straight days.

I would probably ask the question of how to solve this in SQL on stackoverflow.com or with a bit of luck Chat GPT knows that too.

**Comment by MICHAELMUNAVU83** [comment]
-I was thinking for the credits that we give daily , we give them a name eg "Daily Bonus" , so each day a user has just one "Daily Bonus", I will look into the sql stuff and advice.

**Comment by wintermeyer** [comment]
> I will look into the sql stuff and advice

Don't be afraid to solve it with a hammer right now (meaning fetching all of the last 10 days for this donor and filter it with Elixir). We can always make it faster with SQL in the future.
